litmus-portal/graphql-server/manifests/cluster/3b_agents_deployment.yaml

```diff
@@ -6,17 +6,24 @@ metadata:
   namespace: #{AGENT-NAMESPACE}
 data:
   SERVER_ADDR: #{SERVER_ADDR}
-  CLUSTER_ID: #{CLUSTER_ID}
-  ACCESS_KEY: #{ACCESS_KEY}
   AGENT_SCOPE: #{AGENT-SCOPE}
   VERSION: #{VERSION}
   START_TIME: #{START_TIME}
   SKIP_SSL_VERIFY: "#{SKIP_SSL_VERIFY}"
-  CUSTOM_TLS_CERT: "#{CUSTOM_TLS_CERT}"
   IS_CLUSTER_CONFIRMED: #{IS_CLUSTER_CONFIRMED}
   COMPONENTS: |
     DEPLOYMENTS: #{AGENT-DEPLOYMENTS}
 ---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: agent-secret
+  namespace: #{AGENT-NAMESPACE}
+stringData:
+  CLUSTER_ID: #{CLUSTER_ID}
+  ACCESS_KEY: #{ACCESS_KEY}
+  CUSTOM_TLS_CERT: "#{CUSTOM_TLS_CERT}"
```

Comment:
TLS cert is not a secret and can be in the configmap. This is not the server key but the publicly available certificate.

Reply:
It can be in secrets. If you generate a certificate in k8s, the cert-manager stores it in secrets.

Reply:
Everything can be in a secret but that's not my point, the certificate stored by cert-manager contains the private key which is used to sign the certs used by the clients. Compromising that is a security issue but what we are storing here is the publicly available certificate. If you want you can store it in secrets but this information isn't something that needs to be secured any client can connect to the server even without this cert if they skip tls verification.